### PR TITLE
Feature/dispatcher cancel order

### PR DESCRIPTION
**Pull request by domenic99**
#39

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -31,6 +31,16 @@ def update
...
   end
 
+  def cancel
+    if @current_user.try(:instance_of?, Driver)
+      return render json: { 'error' => 'You are not allowed' }, status: 422
+    end
+    order = Order.find(params[:id])
+    order.state = 'canceled'
+    order.save
+    render json: { 'current_order' => 'You have canceled order with id ' + order.id.to_s }
```
Comment (rakvium) [inline app/controllers/orders_controller.rb]: @domenic99 just send updated order as json :wink:

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -27,10 +29,20 @@ def update
...
     order.state = 'in progress'
     order.save
-    render json: { 'current_order' => order.id }
+    render json: { 'current_order' => order }
     send_email_to_client(order.id, order.client_id)
   end
 
+  def cancel
+    if @current_user.try(:instance_of?, Driver)
+      return render json: { 'error' => 'You are not allowed' }, status: 422
```
Comment (rakvium) [inline app/controllers/orders_controller.rb]: @domenic99 this is too harsh :laughing: 
It's better to respond with 'You are not allowed to cancel an order' and status 403 :wink:

File: spec/controllers/orders_controller_spec.rb [hunk 3]
```diff
@@ -1,105 +1,164 @@
...
+    context 'when not logged in' do
+      it 'should return 401' do
+        get :index
+        expect(response).to have_http_status(401)
+      end
+    end
+
+    context 'when logged in as driver' do
+      it 'should return list of orders' do
+        driver = Driver.create FactoryGirl.attributes_for(:driver)
```
Comment (rakvium) [inline spec/controllers/orders_controller_spec.rb]: @domenic99 so what's wrong with `let`? :wink:

Comment (domenic99) [inline spec/controllers/orders_controller_spec.rb]: @rakvium ok I will fix it

**Pull request merged**